informational/req-res-reliability.md

```diff
@@ -0,0 +1,89 @@
+
+### Peers and connection management
+
+#### Pool of reliable service nodes
+Light node should maintain a pool of reliable service nodes for each protocol.
+In case service node fails to serve protocol request from a light node 3 times - light node should drop connection to it and a new service node should be connected and added to the pool instead.
+
+#### Seection of discovered service nodes
+During discovery light node should filter out service nodes based on preferences before establishing connection.
+These preferences might include:
+- [Libp2p multiadresses](https://github.com/libp2p/specs/blob/master/addressing/README.md) of a service node;
+- Waku or libp2p protocols that a service node implements;
+- Wakus shards that a service node is part of;
+
+More details about discovery can be found at [WAKU2 Discovery domain](https://github.com/vacp2p/rfc-index/blob/7b443c1aab627894e3f22f5adfbb93f4c4eac4f6/waku/standards/core/10/waku2.md#discovery-domain) or [RELAY-SHARDING Discovery](https://github.com/waku-org/specs/blob/master/standards/core/relay-sharding.md#discovery).
+
+Examples of filtering:
+- When [Circuit V2](https://github.com/libp2p/specs/blob/master/relay/circuit-v2.md) multiaddresses discovered by a light node - it might avoid connecting such service nodes and wait for service nodes that can be connected directly;
+- When light node discovers service nodes that implement needed Waku protocols - it should prioritize those that implement most recent version of protocol;
+- Light node must connect only to those service nodes that participate in needed shard;
+
+#### Continuous discovery
+Light node must keep information about service nodes up to date.
+For example when a service node is discovered second time,
+we need to be sure to keep connection information up to date in Peer Store.
+
+### Light Push
+
+- While sending message with Light Push - it is advised to use more than 1 peer in order to increase chances of delivering message.
+
+- If sending message is failed to all of the peers - node should try to re-send message after some interval and continue doing so until OK response is received. 
+
+### Filter
+
+- To decrease chances of missing messages a node can initiate more than one subscription through Filter protocol to the same content topic and filter out duplicates. This will increase bandwidth consumption and would depend on the information exchanged under content topic in use.
+
+- In case a node goes offline while having an active subscription - it is important to do ping again right after node appears online. In case ping fails - re-subscribe request should be fired to a new peer.
+
+- While registering Filter subscriptions - it is advised to batch requests for multiple content topics into one in order to reduce amount of queries sent to a node. 
+
+- During creation of a new subscription it can be beneficial to use only new peers to which no subscriptions yet present and not use peers with which Filter already failed.
```

Comment:
I think overall we need to have some sort of local ordering (scoring?) of peers to decide which peers should be used for x based on various criteria

Reply:
yeah, this is kinda [scoring](https://github.com/waku-org/js-waku/issues/2017) or some kind of pre arranged exchange of capabilities of a service node

we don't have any of these so maybe let's initiate discussion if we really want to have such a thing?